src/lib.rs

```diff
@@ -135,7 +136,9 @@ impl State {
 
         while let Some(user) = user_creations_buffer.next().await {
             if let Some(user) = user {
-                self.users.push(user);
+                self.users.insert(user.id().localpart().to_string(), user);
+            } else {
+                println!("couldn't login user");
```

Comment:
same for this `println`

```suggestion
                log::info!("couldn't login user");
```